nvm.sh

```diff
@@ -3679,11 +3802,40 @@ nvm() {
         nvm_sanitize_path nvm_has_colors nvm_process_parameters \
         nvm_node_version_has_solaris_binary nvm_iojs_version_has_solaris_binary \
         nvm_curl_libz_support nvm_command_info nvm_is_zsh nvm_stdout_is_terminal \
+        nvm_get_colors nvm_set_colors nvm_print_color_code nvm_format_help_message_colors \
+        nvm_echo_with_colors \
         >/dev/null 2>&1
       unset NVM_RC_VERSION NVM_NODEJS_ORG_MIRROR NVM_IOJS_ORG_MIRROR NVM_DIR \
-        NVM_CD_FLAGS NVM_BIN NVM_INC NVM_MAKE_JOBS \
+        NVM_CD_FLAGS NVM_BIN NVM_INC NVM_MAKE_JOBS NVM_COLORS NVM_RED_INFO NVM_GREEN_INFO \
+        NVM_BLUE_INFO NVM_CYAN_INFO NVM_MAGENTA_INFO NVM_YELLOW_INFO NVM_BLACK_INFO \
+        NVM_GREY_WHITE_INFO NVM_INSTALLED_COLOR NVM_SYSTEM_COLOR NVM_CURRENT_COLOR \
+        NVM_NOT_INSTALLED_COLOR NVM_DEFAULT_COLOR NVM_LTS_COLOR \
+        CONFIRMATION_MESSAGE NVM_INITIAL_COLOR_INFO \
         >/dev/null 2>&1
     ;;
+    "--set-colors")
+      if [ "${#1}" -eq 5 ] && echo "$1" | grep -E "^[:rRgGbBcCyYmMkKeW:]{1,}$" 1>/dev/null; then
+        nvm_set_colors "$1"
+      else
+        if [ -n "$1" ]; then
+          echo
+          echo "Please pass in five color codes. Choose from: rRgGbBcCyYmMkKeW"
+          nvm help
+          return 1
+        fi
+        if nvm_has_colors; then
+            echo
+            printf "Please use only \033[1;31mvalid color codes\033[0m"
+            echo
```

Comment:
```suggestion
            nvm_err "\nPlease use only \033[1;31mvalid color codes\033[0m\n"
```

Reply:
Since `nvm_err` uses `nvm_echo` and we are using color codes here, we'll add an `nvm_err_with_colors` that calls the `nvm_echo_with_colors` we created to process the escape codes in the arguments passed into it.